msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java

```diff
@@ -188,33 +186,42 @@ public AcquireTokenResult acquireToken(AcquireTokenOperationParameters parameter
         for (int ii = 0; ii < strategies.size(); ii++) {
             final BrokerBaseStrategy strategy = strategies.get(ii);
             try {
-                if (!strategy.hello(parameters)) {
-                    continue;
-                }
-
-                com.microsoft.identity.common.internal.logging.Logger.verbose(
+                com.microsoft.identity.common.internal.logging.Logger.info(
                         TAG + strategyTask.getMethodName(),
                         "Executing with strategy: "
                                 + strategy.getClass().getSimpleName()
                 );
 
+                strategy.hello(parameters);
                 result = strategyTask.perform(strategy, parameters);
                 if (result != null) {
                     break;
                 }
-            } catch (final Exception exception) {
-                if (ii == (strategies.size() - 1)) {
-                    //throw the exception for the last trying of strategies.
-                    if (strategyTask.getTelemetryApiName() != null) {
-                        Telemetry.emit(
-                                new ApiEndEvent()
-                                        .putException(exception)
-                                        .putApiId(strategyTask.getTelemetryApiName())
-                        );
-                    }
-                    throw exception;
+            } catch (final BrokerCommunicationException exception){
+                // continue
+            } catch (final Exception exception){
+                if (strategyTask.getTelemetryApiName() != null) {
+                    Telemetry.emit(
+                            new ApiEndEvent()
+                                    .putException(exception)
+                                    .putApiId(strategyTask.getTelemetryApiName())
+                    );
                 }
+                throw exception;
+            }
+        }
+
+        // This means that we've tried every strategies...
+        if (result == null) {
+            final BrokerCommunicationException exception = new BrokerCommunicationException("MSAL failed to communicate to Broker.");
+            if (strategyTask.getTelemetryApiName() != null) {
+                Telemetry.emit(
+                        new ApiEndEvent()
+                                .putException(exception)
+                                .putApiId(strategyTask.getTelemetryApiName())
```

Comment:
is telemetry api name the same as api id?

Reply:
renamed to getTelemetryApiId() for consistency.